src/sources/github_gist.rs

```diff
@@ -0,0 +1,206 @@
+    }
+
+    fn validate(&self, policy: &BackupPolicy) -> Result<(), crate::Error> {
+        let target: GitHubRepoSourceKind = policy.from.as_str().parse()?;
+
+        match target {
+            GitHubRepoSourceKind::User(u) if u.is_empty() => Err(errors::user(
+                &format!(
+                    "Your 'from' target '{}' is not a valid GitHub username.",
+                    policy.from.as_str()
+                ),
+                "Make sure you provide a valid GitHub username in the 'from' field of your policy.",
+            )),
+            GitHubRepoSourceKind::Gist(gist) if gist.is_empty() => Err(errors::user(
+                &format!(
+                    "Your 'from' target '{}' is not a fully qualified GitHub gist name.",
+                    policy.from.as_str()
+                ),
+                "Make sure you provide a fully qualified GitHub gist name in the 'from' field of your policy.",
+            )),
+            _ => Ok(()),
+        }
+    }
+
+    fn load<'a>(
+        &'a self,
+        policy: &'a BackupPolicy,
+        cancel: &'a AtomicBool,
+    ) -> impl Stream<Item = Result<GitRepo, errors::Error>> + 'a {
+        let target: GitHubRepoSourceKind = policy.from.as_str().parse().unwrap();
+
+        let url = format!(
+            "{}/{}?{}",
+            policy
+                .properties
+                .get("api_url")
+                .unwrap_or(&"https://api.github.com".to_string())
+                .trim_end_matches('/'),
+            target.api_endpoint(self.artifact_kind),
+            policy.properties.get("query").unwrap_or(&"".to_string())
+        )
+        .trim_end_matches('?')
+        .to_string();
+
+        tracing_batteries::prelude::debug!("Calling {} to fetch gists", &url);
+
+        let refspecs = policy
+            .properties
+            .get("refspecs")
+            .map(|r| r.split(',').map(|r| r.to_string()).collect::<Vec<String>>());
+
+        async_stream::try_stream! {
+          if matches!(target, GitHubRepoSourceKind::Gist(_)) {
+            let gist: GitHubGist = self.client.get(&url, &policy.credentials, cancel).await?;
+            yield GitRepo::new(
+              gist.id.as_str(),
+              gist.git_pull_url.as_str(),
+              refspecs.clone())
+                .with_credentials(policy.credentials.clone())
+                .with_metadata_source(&gist);
+          } else {
+            for await gist in self.client.get_paginated(&url, &policy.credentials, cancel) {
+              let gist: GitHubGist = gist?;
+              yield GitRepo::new(
+                gist.id.as_str(),
+                gist.git_pull_url.as_str(),
+                refspecs.clone())
+                  .with_credentials(policy.credentials.clone())
+                  .with_metadata_source(&gist);
+            }
+          }
+        }
+    }
+}
+
+impl GitHubGistSource {
+    #[allow(dead_code)]
+    pub fn with_client(client: GitHubClient, kind: GitHubArtifactKind) -> Self {
```

Comment:
Is the `kind` always `GitHubArtifactKind::Gist` here? If so, maybe we should drop the second parameter here and implement `Default` instead of defining the `::gist()` method?

Reply:
ack